app/models/procedure_revision.rb

```diff
@@ -392,6 +392,8 @@ def conditions_are_valid?
     public_tdcs = types_de_champ_public.to_a
 
     public_tdcs
+      .map { _1.repetition? ? children_of(_1) : [_1] }
+      .flatten
```

Comment:
```suggestion
      .flat_map { _1.repetition? ? children_of(_1) : _1 }
```

(pas sur de mon coup)

Reply:
c'est bon merci !